mm2src/coins/lightning/ln_events.rs

```diff
@@ -124,46 +163,123 @@ fn sign_funding_transaction(
     Transaction::try_from(signed).map_to_mm(|e| OpenChannelError::ConvertTxErr(e.to_string()))
 }
 
+async fn save_channel_closing_details(
+    persister: Arc<LightningPersister>,
+    platform: Arc<Platform>,
+    user_channel_id: u64,
+    reason: String,
+) -> SaveChannelClosingResult<()> {
+    persister.update_channel_to_closed(user_channel_id, reason).await?;
+
+    let channel_details = persister
+        .get_channel_from_sql(user_channel_id)
+        .await?
+        .ok_or_else(|| MmError::new(SaveChannelClosingError::ChannelNotFound(user_channel_id)))?;
+
+    let tx_id = channel_details
+        .funding_tx
+        .ok_or_else(|| MmError::new(SaveChannelClosingError::FundingTxNull))?;
+
+    let tx_hash =
+        H256Json::from_str(&tx_id).map_to_mm(|e| SaveChannelClosingError::FundingTxParseError(e.to_string()))?;
+
+    let funding_tx_bytes = platform
```

Comment:
I guess tx_bytes can be also saved to DB to avoid RPC request here.

Reply:
There are a couple of problems with this approach. 

The first problem is for inbound channels (when I implement channels history for them), there is no event that exposes the funding transaction so I will have to get the funding transaction hash from the exposed `list_channels` function once funding is broadcasted by the counterparty, in this situation this call will be done to get the transaction bytes anyways, so maybe I can make this call conditional depending on the channel type in the future to reduce the number of calls.

The second problem is that if I save the funding_tx_bytes while handling the funding generated event, it's not yet guaranteed that this channel will be opened and the funding tx is broadcasted, since the channel can be closed for a number of reasons before funding is broadcasted. This will lead to the `wait_for_tx_spend` fn to go through the whole wait_until period while doing much more calls instead of only one call to get the transaction bytes that will fail in this case.

Reply:
I see, we can leave this request here then. Thanks for explanation!